bravado/requests_client.py

```diff
@@ -60,13 +68,15 @@ class ApiKeyAuthenticator(Authenticator):
     :param param_in: How to send the API key. Can be 'query' or 'header'.
     """
 
-    def __init__(self, host, api_key, param_name=u'api_key', param_in=u'query'):
+    def __init__(self, host, api_key, param_name='api_key', param_in='query'):
+        # type: (str, str, str, str) -> None
```

Comment:
Same thing here, I expect us to want `param_name` and `param_in` to be unicode strings and not bytestrings? If we're not 100% sure that this is wrong then we should preserve whatever we have.

Reply:
I have to check it more carefully but looks like `param_name` need to have the same type of the values in the headers dictionary.
About `param_in` this is used only internally and is compared with `'headers'`, so the change should be safe.

For `param_in` the fact that is safe does not justify the type change (considering the parameter as part of the public interface) so I'll revert it

Reply:
I've checked the internals of `requests` and looks like `str` refers to `unicode` on python2 ([link|https://github.com/requests/requests/blob/75bdc998e2d430a35d869b2abf1779bd0d34890e/requests/compat.py#L51])

So I'm reverting the `param_name` type change as it is not needed